server/src/main/java/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -1498,7 +1500,7 @@ public boolean checkAndMaintain(final long hostId) {
                 hostInMaintenance = attemptMaintain(host);
             }
         } catch (final NoTransitionException e) {
-            s_logger.debug("Cannot transmit host " + host.getId() + " to Maintenance state", e);
+            s_logger.debug(String.format("Cannot transit %s to Maintenance state", host), e);
```

Comment:
Question / Topic for discussion:

Could exceptions be logged in `debug` instead of `warn` or `error`?

Reply:
I think that `debug` is not ideal indeed at that context, `warn` looks like a good option. `Error` might be good, but I think that it can bring unnecessary alarms for operators.

I think that the current resource state would also be nice to add on the log. I will update it to something as:
```
s_logger.warn(String.format("Cannot transit %s from %s to Maintenance state", host, host.getResourceState()), e);
```